stablehlo/transforms/ShapeLegalizeToStablehlo.cpp

```diff
@@ -559,6 +648,10 @@ struct ShapeLegalizeToStablehloPass
         });
     target->addLegalOp<tensor::CastOp>();
     target->addLegalOp<UnrealizedConversionCastOp>();
+    if (this->legalize_constraints_) {
+      target->addLegalOp<shape::ConstWitnessOp, shape::AssumingOp,
+                         shape::AssumingYieldOp>();
+    }
```

Comment:
IMO it would be cleaner to consolidate the handling of `this->legalize_constraints_` into a single if. If it's true, add the pattern to the set of patterns and add the legal ops.

Reply:
I agree. The current choice is driven by how the code is currenty structured: addition of legal ops handled separately from adding the pattern. Uisng a single if could unify would create an outlier for `legalize_contriants_` case. wdyt?

Reply:
Right, I think both takes are valid, but IMO in this case it would be more readable to have a single `if` since that would make it clearer what happens when `legalize_constraints_` is true. Up to you though.